includes/multi-currency/MultiCurrency.php

```diff
@@ -455,6 +456,61 @@ public function get_switcher_widget_markup( array $instance = [], array $args =
 		return ob_get_clean();
 	}
 
+	/**
+	 * Returns the store's current available, enabled, and default currencies.
+	 *
+	 * @return array
+	 */
+	public function get_store_currencies(): array {
+		return [
+			'available' => $this->get_available_currencies(),
+			'enabled'   => $this->get_enabled_currencies(),
+			'default'   => $this->get_default_currency(),
+		];
+	}
+
+	/**
+	 * Gets the currency settings for a single currency.
+	 *
+	 * @param   string $currency_code The currency code to get settings for.
+	 *
+	 * @return  array The currency's settings.
+	 */
+	public function get_single_currency_settings( string $currency_code ): array {
+		$currency_code = strtolower( $currency_code );
+		return [
+			'exchange_rate_type' => get_option( 'wcpay_multi_currency_exchange_rate_' . $currency_code, 'automatic' ),
+			'manual_rate'        => get_option( 'wcpay_multi_currency_manual_rate_' . $currency_code, null ),
+			'price_rounding'     => get_option( 'wcpay_multi_currency_price_rounding_' . $currency_code, null ),
+			'price_charm'        => get_option( 'wcpay_multi_currency_price_charm_' . $currency_code, null ),
+		];
+	}
+
+	/**
+	 * Updates the currency settings for a single currency.
+	 *
+	 * @param string $currency_code      The single currency code to be updated.
+	 * @param string $exchange_rate_type The exchange rate type setting.
+	 * @param float  $price_rounding     The price rounding setting.
+	 * @param float  $price_charm        The price charm setting.
+	 * @param ?float $manual_rate        The manual rate setting, or null.
+	 *
+	 * @return void
+	 */
+	public function update_single_currency_settings( string $currency_code, string $exchange_rate_type, float $price_rounding, float $price_charm, $manual_rate = null ) {
```

Comment:
Didn't check if we have a protection for that on the UI, but I wonder what will happen if `exchange_rate_type` is `manual` and `manual_rate` is sent as an invalid value, like a string, or `0`?

Reply:
Ok, the string scenario will also be handled by the REST endpoint. The case for 0 is not checked though.

Reply:
It looks like currently the UI allows them to submit whatever they want. With [the check added here](https://github.com/Automattic/woocommerce-payments/pull/6468/files#diff-f5ba94e7313766ad1445e023461ba80603c3abbbe1f8ef0145115dccdda54472R99-R102), it makes sure the value is at least numeric. 

I added in a test check to make sure the number was > 0 before updating the option in the database, but that led to another issue. The data in the component gets updated with what's submitted, so the form retains what is submitted, even though that is not what is in the database. I'll have to update it to throw an exception so that the state in the component is not updated.

Reply:
New exception is thrown in https://github.com/Automattic/woocommerce-payments/pull/6468/commits/cd5c60e9e030148abf340e8bad882ccb3747edb0 tests are included in https://github.com/Automattic/woocommerce-payments/pull/6468/commits/5b51410d700310c1c8133b86f5f2a1a45387be3f . Sorry there were two, I wasn't able to get the tests completed on Friday, but wanted to push my changes that were complete.